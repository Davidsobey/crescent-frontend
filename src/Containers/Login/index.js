import React from "react";
import { connect } from "react-redux";
import { compose } from "redux";
import { submit } from "redux-form";

import { FormControl } from "material-ui/Form";
import { TextField } from "redux-form-material-ui";
import { Field, reduxForm } from "redux-form";

import Card from "../../Components/Card";
import Button from "../../Components/Button";
import UserActions from "../../Actions/UserActions";

import Logo from "./../../Images/LogoText.png";

const validate = values => {
  const errors = {};
  if (!values.password) {
    errors.username = "Required";
  }
  if (!values.email) {
    errors.email = "Required";
  } else if (!/^[A-Z0-9._%+-]+@[A-Z0-9.-]+\.[A-Z]{2,4}$/i.test(values.email)) {
    errors.email = "Invalid email address";
  }
  return errors;
};
class Login extends React.Component {
  submit = values => {
    this.props.dispatch(UserActions.login(values.email, values.password));
  };

  render = () => {
    const { handleSubmit } = this.props;
    const { dispatch } = this.props;
    return (
      <div className="content">
        <form name="form" onSubmit={this.props.handleSubmit(this.submit)}>
          <img src={Logo} className="center-img" width="200px" alt="Crescent" />
          <Card width="500px" title="Lunar Testing Login">
            <br />
            <br />
            <FormControl fullWidth>
              <Field name="email" label="Email Address" component={TextField} />
            </FormControl>
            <FormControl fullWidth>
              <Field
                name="password"
                label="Password"
                type="password"
                component={TextField}
              />
            </FormControl>
            <br />
            <br />
            <div className="alignRight">
              <Button raised color="primary" type="submit">
                Login
              </Button>
            </div>
          </Card>
        </form>
      </div>
    );
  };
}

const mapDispatchToProps = dispatch => {
  return {
    dispatch
  };
};

const withConnect = connect(null, mapDispatchToProps);

const withForm = reduxForm(
  {
    form: "login",
    validate
  },
  Login
);

export default compose(withForm, withConnect)(Login);
