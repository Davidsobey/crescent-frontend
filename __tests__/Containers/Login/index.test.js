import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import configureStore from 'redux-mock-store'; // Smart components

// Component to be tested
import Login from '../../../src/Containers/Login/index.js';


const mockStore = configureStore();
const initialState = {
  LoginReducer:{
    loading: false,
  },
};
const store = mockStore(initialState);

describe('<Login />', () => {
  describe('render()', () => {
    test('renders the component', () => {
      const wrapper = shallow(<Login store={store} />);
      const component = wrapper.dive();

      expect(toJson(component)).toMatchSnapshot();
    });
  });
});