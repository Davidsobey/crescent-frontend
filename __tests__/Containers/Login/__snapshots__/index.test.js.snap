// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Login /> render() renders the component 1`] = `
<Form(Connect(Login))
  anyTouched={false}
  array={
    Object {
      "insert": [Function],
      "move": [Function],
      "pop": [Function],
      "push": [Function],
      "remove": [Function],
      "removeAll": [Function],
      "shift": [Function],
      "splice": [Function],
      "swap": [Function],
      "unshift": [Function],
    }
  }
  arrayInsert={[Function]}
  arrayMove={[Function]}
  arrayPop={[Function]}
  arrayPush={[Function]}
  arrayRemove={[Function]}
  arrayRemoveAll={[Function]}
  arrayShift={[Function]}
  arraySplice={[Function]}
  arraySwap={[Function]}
  arrayUnshift={[Function]}
  asyncValidating={false}
  autofill={[Function]}
  blur={[Function]}
  change={[Function]}
  clearAsyncError={[Function]}
  clearFields={[Function]}
  clearSubmit={[Function]}
  clearSubmitErrors={[Function]}
  destroy={[Function]}
  destroyOnUnmount={true}
  dirty={false}
  dispatch={[Function]}
  enableReinitialize={false}
  focus={[Function]}
  forceUnregisterOnUnmount={false}
  form="login"
  getFormState={[Function]}
  initialize={[Function]}
  initialized={false}
  invalid={false}
  keepDirtyOnReinitialize={false}
  persistentSubmitErrors={false}
  pristine={true}
  pure={true}
  registerField={[Function]}
  reset={[Function]}
  resetSection={[Function]}
  setSubmitFailed={[Function]}
  setSubmitSucceeded={[Function]}
  shouldAsyncValidate={[Function]}
  shouldError={[Function]}
  shouldValidate={[Function]}
  shouldWarn={[Function]}
  startAsyncValidation={[Function]}
  startSubmit={[Function]}
  stopAsyncValidation={[Function]}
  stopSubmit={[Function]}
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
  storeSubscription={
    Subscription {
      "listeners": Object {
        "clear": [Function],
        "get": [Function],
        "notify": [Function],
        "subscribe": [Function],
      },
      "onStateChange": [Function],
      "parentSub": undefined,
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "unsubscribe": [Function],
    }
  }
  submit={[Function]}
  submitFailed={false}
  submitSucceeded={false}
  submitting={false}
  syncErrors={Object {}}
  syncWarnings={Object {}}
  touch={[Function]}
  touchOnBlur={true}
  touchOnChange={false}
  unregisterField={[Function]}
  untouch={[Function]}
  updateSyncErrors={[Function]}
  updateSyncWarnings={[Function]}
  updateUnregisteredFields={false}
  valid={true}
  validExceptSubmit={true}
  validate={[Function]}
  values={Object {}}
/>
`;
